base/sysinfo.jl

```diff
@@ -570,6 +573,45 @@ function isexecutable(path::String)
 end
 isexecutable(path::AbstractString) = isexecutable(String(path))
 
+"""
+    Sys.isreadable(path::String)
+
+Return `true` if the given `path` has readable permissions.
+"""
+function isreadable(path::String)
+    # We use `access()` and `R_OK` to determine if a given path is
+    # executable by the current user.  `R_OK` comes from `unistd.h`.
```

Comment:
nit: for all of these, update this comment
```suggestion
    # readable by the current user.  `R_OK` comes from `unistd.h`.
```